source/extensions/tracers/opencensus/opencensus_tracer_impl.cc

```diff
@@ -232,6 +232,39 @@ Tracing::SpanPtr Span::spawnChild(const Tracing::Config& /*config*/, const std::
 
 void Span::setSampled(bool sampled) { span_.AddAnnotation("setSampled", {{"sampled", sampled}}); }
 
+class GoogleUserProjHeaderInterceptor : public grpc::experimental::Interceptor {
+public:
+  GoogleUserProjHeaderInterceptor(const std::string& project_id) : project_id_(project_id) {}
+
+  virtual void Intercept(grpc::experimental::InterceptorBatchMethods* methods) {
+    if (methods->QueryInterceptionHookPoint(
```

Comment:
@g-easy does this look ok?
It is being directly pushed into an intermediary version of envoy in istio.

Reply:
This should not be needed in envoy master after https://github.com/grpc/grpc/issues/21225 is resolved.

Reply:
This seems reasonable to me but I haven't dealt with interceptors before.